Code/autopkglib/URLGetter.py

```diff
@@ -62,10 +62,19 @@ def curl_binary(self):
 
         raise ProcessorError("Unable to locate or execute any curl binary")
 
+    def prepare_curl_cmd(self):
+        """Assemble basic curl command and return it."""
+        return [self.curl_binary(), "--compressed", "--location"]
+
+    def add_curl_headers(self, curl_cmd, headers):
+        """Add headers to curl_cmd"""
+        if headers:
+            for header, value in headers.items():
+                curl_cmd.extend(["--header", "%s: %s" % (header, value)])
```

Comment:
Use `.format()` instead of string interpolation here, because it'll be easier to transition in the future.

Reply:
Ok. Will do.

Reply:
Pushed 5 new commits replacing string interpolation with format in `URLGetter` and subclasses.